src/MainWindow.vala

```diff
@@ -454,7 +454,8 @@ public class Torrential.MainWindow : Gtk.Window {
 
         var clipboard = Gtk.Clipboard.get (Gdk.SELECTION_CLIPBOARD);
         string? contents = clipboard.wait_for_text ();
-        if (contents != null || contents != "") {
+        if (contents != null && contents.has_prefix ("magnet:")) {
+
```

Comment:
```suggestion

```

You've added an extra blank line here, that isn't necessary, but otherwise looks good, thanks!